lib/index.js

```diff
@@ -337,31 +337,51 @@ function Postgres(a, b) {
   }
 
   function listen(channel, fn) {
+    const listener = getListener();
+
     if (channel in listeners) {
       listeners[channel].push(fn)
-      return Promise.resolve(channel)
+      return Promise.resolve(listener.result)
     }
 
     listeners[channel] = [fn]
-    return query({ raw: true }, getListener(), 'listen ' + escape(channel))
+    return query({ raw: true }, listener.conn, 'listen ' + escape(channel))
+      .then((response) => {
+        response.unlisten = unlisten
+        listener.result = response
+        return response
+      })
+
+    function unlisten() {
+      const index = listeners[channel].indexOf(fn)
+      if (index !== -1)
+        listeners[channel].splice(index, 1)
+      if (!listeners[channel].length) {
+        delete listeners[channel]
```

Comment:
Won't this mean that the next calls to `unlisten` will raise an exception, since after the first call `listeners[channel]` is `undefined`?

Reply:
good point!

Reply:
good point!

Reply:
Fixed